WcaOnRails/app/views/competitions/_registration_requirements.html.erb

```diff
@@ -0,0 +1,59 @@
+<% if @competition.use_wca_registration? %>
```

Comment:
Have we considered what will happen here with upcoming competitions on the day we merge this code up? Looking at the migration below, here's what I think will happen:

- `on_the_spot_registration` will be set to NULL
- `on_the_spot_entry_fee_lowest_denomination` will be set to 0
- `refund_policy_percentage` will be set to 0
- `refund_policy_limit_date` will be set to NULL
- `guests_entry_fee` will be set to 0

I think that means we'll render the `competitions.competition_info.no_on_the_spot_registration` message, the `competitions.competition_info.guests_free` message, and maybe the `competitions.competition_info.no_refunds` message (it's hard to tell what will happen with that one, because I don't know what the `if @competition.refund_policy_percentage?` statement you wrote will do if `refund_policy_percentage` is equal to 0. I suspect that Ruby would treat that as true...

Unfortunately, I think we need to figure all this out before we go live.I think we need to update this `_registration_requirements.html.erb` partial to handle missing information. In order to do this, we'd need to update our database migration so it's possible for us to record the "we don't know" state for some of these things. For instance, currently `guests_entry_fee` is not NULL-able and has a default of 0, so if we run the current migration, it will say all past (and upcoming!) competitions have free guest entry, which we definitely don't want.

Another approach would be to manually go and try to collect all this structured data for past competitions, but that would be a ton of work, and I don't think a good use of anyone's time.

Hopefully this makes sense! Feel free to hit me up on slack if you have any questions.

Reply:
I believe that going live with this won't break anything. At least, when I go to a page of a past competition on local, I don't get any error. The bad thing is that this will display FALSE information for past competitions, and that doesn't sound good :(